### ISSUE TITLE
E203: false positive in f-string using formatting

### ISSUE THREAD

**Issue by JaRoSchm** [astral-sh/ruff#12023]
Hi,

this example raises E203 (ruff 0.4.10) although it clearly shouldn't in my opinion:
```python
x = 3.14159
print(f"{x = :.2f}")
```

Output of `ruff check --select E203 --preview`:
```
test.py:2:13: E203 [*] Whitespace before ':'
  |
1 | x = 3.14159
2 | print(f"{x = :.2f}")
  |             ^ E203
  |
  = help: Remove whitespace before ':'
```

However, removing the whitespace changes the output from `x = 3.14` to `x =3.14`.

**Comment by dhruvmanila** [comment]
Yeah, I agree. I remember updating the rules to avoid checking inside f-strings when I added support for it, I guess I must've missed this one. Thank you for reporting!
